src/components/UserSelector.tsx

```diff
@@ -1,16 +1,62 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { getUsers } from '../api/ClientFunctions';
+import { User } from '../types/User';
+import classNames from 'classnames';
+
+interface Props {
+  currUser: User | null;
+  isDdActive: boolean;
+  setPrevUser: (user: User | null) => void;
+  setCurrUser: (user: User | null) => void;
+  setIsDdActive: (value: boolean) => void;
+}
+
+export const UserSelector: React.FC<Props> = ({
+  currUser,
+  isDdActive,
+  setPrevUser,
+  setCurrUser,
+  setIsDdActive,
+}) => {
+  const [users, setUsers] = useState<User[]>([]);
+
+  // #region handlers
+
+  const buttonOnClickHandler = () => {
+    if (isDdActive) {
+      setIsDdActive(false);
+
+      return;
+    }
+
+    setIsDdActive(true);
+  };
+
+  const selectUserHandler = (user: User) => {
+    setPrevUser(currUser);
+    setCurrUser(user);
+  };
+
+  // #endregion
+
+  useEffect(() => {
+    getUsers().then(_users => setUsers(_users));
```

Comment:
Consider adding error handling for the `getUsers` API call to manage potential errors and provide feedback to the user if the request fails.